### PR TITLE
Added a new option that allows you to increase the system image size for qemu

### PR DESCRIPTION
**Pull request by BrianOtto**
I have added a new `--image-size` option for qemu, as described in #551.

I feel like `--increase-image-size` is a more accurate name for what the option does, but it seemed a little long to type. Let me know if you think it should be changed, or if you have any better ideas.

### CODE REVIEW

**Comment by drebrez** [comment]
@BrianOtto Nice feature, I agree with you that `--image-size` might be confusing, since the value specified gets added to the actual image.
In theory we can have both arguments, `--increase-image-size` that increase the image by the amount specified and `--image-size` which sets the image to that specific size, it will only change the `truncate` command to have the `+` or not.
By setting the specific size we should add a check to not allow to specify a smaller value than the actual image size, this because the initramfs script which doesn't handle the shrink of the partitions, we can think about it in the future.
@BrianOtto and @ollieparanoid , what you think about having both arguments?

**Comment by BrianOtto** [comment]
@drebrez I am okay with adding two arguments.

Another alternative is maybe we just make it behave as you would expect, but with the safety check @drebrez mentioned, i.e.
> `--image-size` which sets the image to that specific size, it will only change the truncate command to have the + or not.
By setting the specific size we should add a check to not allow to specify a smaller value than the actual image size, this because the initramfs script which doesn't handle the shrink of the partitions, we can think about it in the future.

If we do it this way, then you get the additional benefit of not accidentally increasing the image size when running it again from your command line history. I actually ran into that a few times when testing this update :)

Having said that, there is one issue we could run into. If I set the size to say 1GB on one run, but then decide to decrease it on the next run, how do I know the original image size now? On the first run, we know this by doing a os.path.getsize(img_path), but on the 2nd run this will be 1GB (i.e. the **increased size**). Is the original image size (base image + size of all packages) stored in cache somewhere?

**Comment by ollieparanoid** [comment]
> If we do it this way, then you get the additional benefit of not accidentally increasing the image size when running it again from your command line history. I actually ran into that a few times when testing this :)

That's a very good argument! So If I understood you right, you're proposing that we only use one argument, and that sets the absolute size? That sounds like a great solution to me!

About the problem you mentioned, I would never allow to shrink it again. Because as soon as you boot it after resizing, you have no guarantee that there hasn't been any data written to the increased space.
After all, re-building the image is just less than a minute of zap and install away :)

**Comment by BrianOtto** [comment]
> So If I understood you right, you're proposing that we only use one argument, and that sets the absolute size?

Yes, that's correct.

> About the problem you mentioned, I would never allow to shrink it again. Because as soon as you boot it after resizing, you have no guarantee that there hasn't been any data written to the increased space.
After all, re-building the image is just less than a minute of zap and install away :)

Good point, yes, you wouldn't want to shrink it... and it is easy to re-build too!

Great, thanks everyone! I will make this change.

**Comment by BrianOtto** [comment]
The updates have been made. Let me know what you think.

File: pmb/qemu/run.py [hunk 1]
```diff
@@ -154,6 +155,42 @@ def run(args):
     command = qemu_command(args, arch, device, img_path)
 
     logging.info("Command: " + " ".join(command))
+
+    if args.image_size:
+        # current image size in bytes
```
Comment (ollieparanoid) [inline pmb/qemu/run.py]: The new block is quite large, so I would move it to an extra function, like this:
```python
def resize_image(args, img_size_new):
    """
    (Description of the function)

    :param img_size_new: new image size in M or G
    """
    
    # ...


def run(args):
    # ...
    if args.image_size:
        resize_image(args, args.image_size)
    # ...

```

**Comment by BrianOtto** [comment]
The `--image-size` logic has been moved to a new function now.

**Comment by ollieparanoid** [comment]
I like how well commented everything is! Thank you very much, tested and working. :ship:

**Pull request merged**